Specs/Core/EventSpec.js

```diff
@@ -67,6 +67,29 @@ defineSuite([
         expect(event.numberOfListeners).toEqual(0);
     });
 
+    it('can remove multiple listeners within a callback', function() {
+        var listeners = [];
+
+        function addListener(index) {
+            var remove;
+            var listener = function() {
+                if (index % 2 === 0) {
+                    remove();
```

Comment:
This test is a little hard to follow. Can you try structuring it a little more linearly like the other tests?

Reply:
Sure